compiler/lcalc/optimizations.ml

```diff
@@ -14,59 +14,116 @@
 open Utils
 open Ast
 
-let rec peephole_expr (e : expr Pos.marked) : expr Pos.marked Bindlib.box =
+let ( let+ ) x f = Bindlib.box_apply f x
+
```

Comment:
As I mentionned, this rewrite of `LCalc.Optimizations` is not feature-preserving since it misses some optimizations. For instance the new code does not optimize `(false && false) || true`.

Reply:
I propose to just remove the code inside optimization.ml, and to move it to an other PR/just drop it. The beta-reduction optimization is not working either.

One over way could be to add some sort of "size" of the ast, and to use this size to keep optimizing the program.

Reply:
Ok, can you file an issue about rewriting optimization passes for `Lcalc` ?

Reply:
I think I have a fix. Iota/beta transformation look like to be working

Reply:
There appears to a a bug, see my comment on the PR.